includes/admin/class-wcmypa-admin.php

```diff
@@ -906,6 +918,85 @@ private function printDeliveryDate(DeliveryOptions $deliveryOptions): void
     }
 
     /**
+     * Output the delivery date if there is a date and the show delivery day setting is enabled.
+     *
+     * @param DeliveryOptions $deliveryOptions
+     * @param bool            $mail
+     *
+     * @throws \Exception
+     */
+    private function getConfirmationData(DeliveryOptions $deliveryOptions, bool $mail): void
+    {
+        $signatureTitle     = (new WCMP_Checkout)->getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_SIGNATURE_TITLE);
+        $onlyRecipientTitle = (new WCMP_Checkout)->getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_ONLY_RECIPIENT_TITLE);
+
+        if (! $deliveryOptions->getCarrier()) {
+            return;
+        }
+
+        if ($deliveryOptions->getDeliveryType() === AbstractConsignment::DELIVERY_TYPE_PICKUP_NAME) {
+            $selectedDeliveryOptions = [
+                __("Delivery type:", "woocommerce-myparcel")   => WCMP_Data::getDeliveryTypesHuman()[$deliveryOptions->getDeliveryType()],
+                __("Pickup location:", "woocommerce-myparcel") =>
+                    $deliveryOptions->getPickupLocation()->getLocationName() . '<br>' .
+                    $deliveryOptions->getPickupLocation()->getStreet() . ' ' .
+                    $deliveryOptions->getPickupLocation()->getNumber() . '<br>' .
+                    $deliveryOptions->getPickupLocation()->getPostalCode() . ' ' .
+                    $deliveryOptions->getPickupLocation()->getCity()
+            ];
+
+            printf($this->setConfirmationData($selectedDeliveryOptions, $mail));
+
+            return;
+        }
+
+
+        $selectedDeliveryOptions = [
+            __("Delivery type:", "woocommerce-myparcel") => WCMP_Data::getDeliveryTypesHuman()[$deliveryOptions->getDeliveryType()],
+            __("Date:", 'woocommerce')                   => wc_format_datetime(new WC_DateTime($deliveryOptions->getDate())),
+            __("Extra options:", "woocommerce-myparcel") =>
+                ($deliveryOptions->getShipmentOptions()->hasSignature() ? $signatureTitle . '<br>' : null) .
+                ($deliveryOptions->getShipmentOptions()->hasOnlyRecipient() ? $onlyRecipientTitle : null)
+        ];
+
+        printf($this->setConfirmationData($selectedDeliveryOptions, $mail));
+    }
+
+    /**
+     * @param array $options
```

Comment:
in plaats van alleen `array` liever opschrijven wat erin zit, dus bijvoorbeeld `string[]` of `array[]`

Reply:
done